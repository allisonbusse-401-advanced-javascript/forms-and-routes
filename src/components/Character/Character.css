.Character {

  button {
    text-transform: uppercase;
    padding: 5px;
    background-color: darkcyan;
    color: white;
    font-size: 16px;
    display: block;
    border-radius: 5px;
  }

  img {
    border-radius: 8px;
  }

  p {
    font-family: Arial, Helvetica, sans-serif;
  }
}

h2 {
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  color: darkslategray;
}