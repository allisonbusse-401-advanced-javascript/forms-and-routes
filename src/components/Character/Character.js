import React from 'react';
import styles from './Character.css';
import PropTypes from 'prop-types';

const Character = ({ photo, name, handleClick }) => {
  return (
    <div className={styles.Character}>
      <img src={photo}></img>
      <p>{name}</p>
      <button onClick={handleClick}>Get a New Character</button>
    </div>
  );
};

Character.propTypes = {
  photo: PropTypes.string.isRequired,
  name: PropTypes.string,
  handleClick: PropTypes.func.isRequired
};

export default Character;

