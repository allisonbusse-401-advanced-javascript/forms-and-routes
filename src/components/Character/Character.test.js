import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a character', () => {
    const handleClick = () => {
      // eslint-disable-next-line no-console
      console.log('hello');
    };
    const wrapper = shallow(
      <Character 
        name="Test Title"
        photo="test image"
        handleClick={handleClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  
});
