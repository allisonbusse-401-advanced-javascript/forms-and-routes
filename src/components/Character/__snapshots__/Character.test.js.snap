// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<div
  className="Character"
>
  <img
    src="test image"
  />
  <p>
    Test Title
  </p>
  <button
    onClick={[Function]}
  >
    Get a New Character
  </button>
</div>
`;
